tokens/contracts/erc-1155/ERC1155Base.sol

```diff
@@ -35,5 +35,56 @@ abstract contract ERC1155Base is OwnableUpgradeable, ERC1155DefaultApproval, ERC
         return super.supportsInterface(interfaceId);
     }
 
+    function burn(address account, uint256 id, uint256 value) public virtual override {
+        if (_isExist(id)) {
+            _burnExisted(account, id, value);
+        } else {
+            _burnLazy(account, id, value);
+        }
+    }
+
+    function _burnLazy(address account, uint256 id, uint256 value) internal {
+        require(account == _msgSender(), "ERC1155: caller is not burner");
+        address minter = address(id >> 96);
+        require(minter == _msgSender(), "ERC1155: caller is not token owner");
```

Comment:
do we need this condition? it's checked in _burnExisted

Reply:
agree simplify logic